### ISSUE TITLE
Error: Creating too many calendar events

### ISSUE THREAD

**Issue by amytencza** [derekantrican/GAS-ICS-Sync#46]
On first run I am getting the error: "You have been creating or deleting too many calendars or calendar events in a short time. Please try again later. at main(Code:190)"

**Comment by derekantrican** [comment]
@amytencza Is your issue related to #39?

**Comment by amytencza** [comment]
No. The script isn't timing out. It stops running somewhere between 9 and 30 seconds with a "Failure" status. The logs give the error above. I believe it's a limit on google calendar, not a timeout issue. Perhaps a small pause between calendar updates (although it will increase script run time).....?

**Comment by amytencza** [comment]
although note that i kept running it and it eventually got through them all.....

**Comment by jonas0b1011001** [comment]
Same error as mentioned in #25 
Maybe try adding "Utilities.sleep(500);" as suggested by Derek.

Also do you have set addAlerts = true ?

**Comment by amytencza** [comment]
looks like if we add the sleep it'll time out per the comments in that other issue. however, i think i'm past that error now. 

i do have addAlerts = true;

**Comment by amytencza** [comment]
same error

**Comment by tanoabeleyra** [comment]
@amytencza I don't know what's your case but I was having the same problem with a calendar that has lot of events and given that I didn't care about old events I did this:

```
diff --git a/Code.gs b/Code.gs
index 5fe7cc4..107da02 100644
--- a/Code.gs
+++ b/Code.gs
@@ -165,8 +165,9 @@ function main(){
   //------------------------ Add events to calendar ----------------
   if (addEventsToCalendar){
     Logger.log("Checking " + events.length + " Events for creation")
+    var start = new Date(2019,00,01);
     for each (var event in events){
-      if (calendarFids.indexOf(event.id) == -1){
+      if (calendarFids.indexOf(event.id) == -1 && event.startTime > start){
         var resultEvent;
         if (event.isAllDay){
           resultEvent = targetCalendar.createAllDayEvent(event.title, 

```

**Comment by derekantrican** [comment]
### Closing as this is fixed by the latest release: [v5.0](https://github.com/derekantrican/GAS-ICS-Sync/releases/tag/5.0)

*If you find that there are still bugs, please comment on this issue and let us know*
